src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs

```diff
@@ -195,7 +207,55 @@ private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(
             }
         }
 
-        return results.Values.ToList();
+        return results.Values;
+    }
+
+    internal static Dictionary<int, ApiResponseType> ReadResponseMetadata(
```

Comment:
Is it worthwhile to figure out a way to codeshare with the implementation in `EndpointMetadataApiDescriptionProvider` here?

https://github.com/dotnet/aspnetcore/blob/ef6286fb88a1eb5e73af0c1bf27e0a2b9afc1b7d/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L394-L396